File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/experience/mapper/ExperienceContentMapper.java
package com.bamyanggang.persistence.experience.mapper;

import com.bamyanggang.domainmodule.domain.experience.aggregate.ExperienceContent;
import com.bamyanggang.persistence.experience.jpa.entity.ExperienceContentJpaEntity;
import org.springframework.stereotype.Component;

@Component
public class ExperienceContentMapper {
    public ExperienceContentJpaEntity toJpaEntity(ExperienceContent experienceContent) {
        return new ExperienceContentJpaEntity(
                experienceContent.getId(),
                experienceContent.getQuestion(),
                experienceContent.getAnswer()
        );
    }

    public ExperienceContent toDomainEntity(ExperienceContentJpaEntity experienceContentJpaEntity) {
        return ExperienceContent.Companion.toDomain(
                experienceContentJpaEntity.getExperienceContentId(),
                experienceContentJpaEntity.getQuestion(),
                experienceContentJpaEntity.getAnswer()
        );
    }
}

File: Infrastructure-Module/persistence/src/main/java/com/bamyanggang/persistence/experience/mapper/ExperienceContentMapper.kt
package com.bamyanggang.persistence.experience.mapper

import com.bamyanggang.domainmodule.domain.experience.aggregate.ExperienceContent
import com.bamyanggang.persistence.experience.jpa.entity.ExperienceContentJpaEntity
import org.springframework.stereotype.Component

@Component
class ExperienceContentMapper(
    private val experienceMapper: ExperienceMapper
) {

    fun toDomain(experienceContentJpaEntity: ExperienceContentJpaEntity): ExperienceContent {
        return ExperienceContent(
            experienceContentJpaEntity.id,
            experienceContentJpaEntity.question,
            experienceContentJpaEntity.answer,
            experienceMapper.toDomainEntity(experienceContentJpaEntity.experience)
        )
    }

    fun toJpaEntity(experienceContent: ExperienceContent) : ExperienceContentJpaEntity {
        return ExperienceContentJpaEntity.of(
            experienceContent.id,
            experienceContent.question,
            experienceContent.answer,
            experienceMapper.toJpaEntity(experienceContent.experience)
        )
    }
}